Lib9c/Model/Character/Enemy.cs

```diff
@@ -49,18 +48,23 @@ private void PostConstruction()
         protected override void OnDead()
         {
             base.OnDead();
-            var player = (Player) Targets[0];
+            var player = (Player)Targets[0];
             player.RemoveTarget(this);
         }
 
         protected sealed override void SetSkill()
         {
             base.SetSkill();
 
-            var dmg = (int) (ATK * 0.3m);
-            var skillIds = StageSimulator.EnemySkillSheet.Values.Where(r => r.characterId == RowData.Id)
-                .Select(r => r.skillId).ToList();
-            var enemySkills = Simulator.SkillSheet.Values.Where(r => skillIds.Contains(r.Id))
+            var dmg = (int)(ATK * 0.3m);
+            var skillIds = ((IStageSimulator)Simulator)
```

Comment:
I'm not an expert on this project but the `Enemy` class requires the `Simulator` to be `IStageSimulator`. In my opinion, it seems better to check the type in the constructor like before because it prevents to create `Enemy` object with `Simulator` not typed `IStageSimulator`.

Reply:
@moreal It seems be possible to reduce member field when I was refactoring the `Enemy` class.
But I also agree to validate arguments in constructor.
Thanks!

cc @ipdae